hw/top_earlgrey/dv/chip_rom_tests.hjson

```diff
@@ -41,6 +41,261 @@
+      ]
+      en_run_modes: ["sw_test_mode_rom"]
+      run_opts: [
+        "+sw_test_timeout_ns=50_000_000",
+        "+use_otp_image=OtpTypeCustom",
+        "+bypass_alert_ready_to_end_check=1",
+      ]
+      reseed: 1
+      run_timeout_mins: 120
+    }
+    {
+      name: rom_e2e_sigverify_always_a_bad_b_nothing_dev
+      // We can use the same vseq as the `*_a_bad_b_bad_*` tests above, as we expect the same ROM boot fault value.
+      uvm_test_seq: chip_sw_rom_e2e_sigverify_always_a_bad_b_bad_vseq
+      sw_images: [
+        "//sw/device/silicon_creator/rom/e2e:empty_test_slot_a_corrupted:1:ot_flash_binary:signed:fake_dev_key_0:no_sw_log_db",
+        "//sw/device/silicon_creator/rom/e2e:otp_img_sigverify_always_dev:4",
+      ]
+      en_run_modes: ["sw_test_mode_rom"]
+      run_opts: [
+        "+sw_test_timeout_ns=50_000_000",
+        "+use_otp_image=OtpTypeCustom",
+        "+bypass_alert_ready_to_end_check=1",
+      ]
+      reseed: 1
+      run_timeout_mins: 120
+    }
+    {
+      name: rom_e2e_sigverify_always_a_bad_b_nothing_prod
+      // We can use the same vseq as the `*_a_bad_b_bad_*` tests above, as we expect the same ROM boot fault value.
+      uvm_test_seq: chip_sw_rom_e2e_sigverify_always_a_bad_b_bad_vseq
+      sw_images: [
+        "//sw/device/silicon_creator/rom/e2e:empty_test_slot_a_corrupted:1:ot_flash_binary:signed:fake_prod_key_0:no_sw_log_db",
+        "//sw/device/silicon_creator/rom/e2e:otp_img_sigverify_always_prod:4",
+      ]
+      en_run_modes: ["sw_test_mode_rom"]
+      run_opts: [
+        "+sw_test_timeout_ns=50_000_000",
+        "+use_otp_image=OtpTypeCustom",
+        "+bypass_alert_ready_to_end_check=1",
+      ]
+      reseed: 1
+      run_timeout_mins: 120
+    }
+    {
+      name: rom_e2e_sigverify_always_a_bad_b_nothing_prod_end
+      // We can use the same vseq as the `*_a_bad_b_bad_*` tests above, as we expect the same ROM boot fault value.
+      uvm_test_seq: chip_sw_rom_e2e_sigverify_always_a_bad_b_bad_vseq
+      sw_images: [
+        "//sw/device/silicon_creator/rom/e2e:empty_test_slot_a_corrupted:1:ot_flash_binary:signed:fake_prod_key_0:no_sw_log_db",
+        "//sw/device/silicon_creator/rom/e2e:otp_img_sigverify_always_prod_end:4",
+      ]
+      en_run_modes: ["sw_test_mode_rom"]
+      run_opts: [
+        "+sw_test_timeout_ns=50_000_000",
+        "+use_otp_image=OtpTypeCustom",
+        "+bypass_alert_ready_to_end_check=1",
+      ]
+      reseed: 1
+      run_timeout_mins: 120
+    }
+    {
+      name: rom_e2e_sigverify_always_a_bad_b_nothing_rma
+      // We can use the same vseq as the `*_a_bad_b_bad_*` tests above, as we expect the same ROM boot fault value.
+      uvm_test_seq: chip_sw_rom_e2e_sigverify_always_a_bad_b_bad_vseq
+      sw_images: [
+        "//sw/device/silicon_creator/rom/e2e:empty_test_slot_a_corrupted:1:ot_flash_binary:signed:fake_prod_key_0:no_sw_log_db",
+        "//sw/device/silicon_creator/rom/e2e:otp_img_sigverify_always_rma:4",
```

Comment:
Is there a standard list of all of the different kinds of custom OTP images? Seems like that list could quickly grow pretty large with various different combinations of settings.

Reply:
This OTP image is defined-in (and built by) Bazel [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/e2e/BUILD;drc=344398756af305af675e76aa081f6bf16dc8c3bc;l=2328). The `OtpTypeCustom` plusarg just says "Bazel please build the OTP image based on the provided Bazel target label in the `sw_images` list". This enables us to use the same OTP image across all platforms (DV, Verilator, FPGA), as Bazel is the sole source of truth. So there is no finite list of Bazel OTP images, developers are free to use the rules in `rules/otp.bzl` to define any OTP images they would like.

Reply:
Ok, got it.